cluster/aws/util.sh

```diff
@@ -435,6 +449,57 @@ function ensure-master-pd {
   fi
 }
 
+# Gets master ELB, if exists
+# Sets MASTER_ELB_ID
+# Note that for ELB, the name == the ID, but we gloss over that:
+# to be consistent, we only set the ID if the load balacner exists.
+function find-master-elb {
+  # Sadly ELBs don't support querying by tag
+  # We can tag them, but we can't query on tags (easily)
+  # It's also tricky to tolerate the LoadBalancerNotFound error
+  # For now, we list them all and match on name
+  if [[ -z "${MASTER_ELB_ID:-}" ]]; then
+    local elb_ids=$(get_elbs_in_vpc ${VPC_ID})
+    if [[ -n "${elb_ids}" ]]; then
+      for elb_id in ${elb_ids}; do
+        if [[ "${elb_id}" == "${MASTER_ELB_NAME}" ]]; then
+          MASTER_ELB_ID=${elb_id}
+        fi
+      done
+    fi
+  fi
+
+  if [[ -n "${MASTER_ELB_ID:-}" ]]; then
+    if [[ -z "${MASTER_ELB_DNSNAME:-}" ]]; then
+      MASTER_ELB_DNSNAME=`aws elb describe-load-balancers \
+                                  --load-balancer-names ${MASTER_ELB_ID} \
+                                  --query LoadBalancerDescriptions[].DNSName`
+    fi
+  fi
+}
+
+# Gets or creates master ELB
+# Sets MASTER_ELB_ID
+function ensure-master-elb {
+  find-master-elb
+
+  if [[ -z "${MASTER_ELB_ID:-}" ]]; then
+    echo "Creating master ELB"
+    aws elb create-load-balancer \
+	    --load-balancer-name ${MASTER_ELB_NAME} \
+	    --listeners Protocol=TCP,LoadBalancerPort=443,InstanceProtocol=TCP,InstancePort=443 \
+	    --subnets ${SUBNET_ID} \
+	    --security-groups ${MASTER_ELB_SG_ID} \
+	    --tags Key=KubernetesCluster,Value=${CLUSTER_ID} \
+	           Key=kubernetes.io/role,Value=master-elb > ${LOG}
+    find-master-elb
+    if [[ -z "${MASTER_ELB_ID:-}" ]]; then
+      echo "Created master ELB, but was then unable to find it"
```

Comment:
You mean in unhealthy state, right?  I commented on this around line 1091.

It is annoying, I agree.  I'm not entirely sure what the right behaviour is - what do you suggest @therc ?  Reduce the delay before marking a node as healthy again?

Currently the cluster will take a few extra minutes to start up right now (when running with ELB), but otherwise everything works :-)